onnxruntime/core/providers/cpu/math/einsum_utils/einsum_compute_preprocessor.h

```diff
@@ -190,11 +194,15 @@ class EinsumComputePreprocessor final {
 
   // Allocator to use for ad-hoc tensor buffer allocation
   AllocatorPtr allocator_;
-};
 
-// This method does the heavy-lifting compute portion of Einsum Compute()
-template <typename T>
-Status EinsumTypedComputeProcessor(OpKernelContext* context, AllocatorPtr allocator,
-                                   EinsumComputePreprocessor& einsum_compute_preprocessor);
+  // Device specific diagonal function
+  EinsumOp::DeviceHelpers::Diagonal device_diagonal_func_;
+
+  // Device specific transpose function
+  EinsumOp::DeviceHelpers::Transpose device_transpose_func_;
+
+  // Holds CUDA assets required for CUDA ops that need to be executed
+  void* einsum_cuda_assets_;
+};
```

Comment:
nit: this could be used by any EP (e.g. DML may want to do that) and not just the CUDA EP so the name could be more generic.

Reply:
Right. named them `einsum_ep_assets_`